src/main/java/org/broadinstitute/dsm/model/BSPKit.java

```diff
@@ -0,0 +1,150 @@
+                            }
+                        }
+                        else {
+                            triggerDDP(conn, bspKitQueryResult, firstTimeReceived, kitLabel);
+                        }
+                    }
+                }
+                else {
+                    triggerDDP(conn, bspKitQueryResult, firstTimeReceived, kitLabel);
+                }
+
+                String bspParticipantId = bspKitQueryResult.getBspParticipantId();
+                String bspSampleId = bspKitQueryResult.getBspSampleId();
+                String bspMaterialType = bspKitQueryResult.getBspMaterialType();
+                String bspReceptacleType = bspKitQueryResult.getBspReceptacleType();
+                int bspOrganism;
+                try {
+                    bspOrganism = Integer.parseInt(bspKitQueryResult.getBspOrganism());
+                }
+                catch (NumberFormatException e) {
+                    throw new RuntimeException("Organism " + bspKitQueryResult.getBspOrganism() + " can't be parsed to integer", e);
+                }
+
+                logger.info("Returning info for kit w/ label " + kitLabel + " for " + bspKitQueryResult.getInstanceName());
+                return new BSPKitInfo(bspKitQueryResult.getBspCollection(),
+                        bspOrganism,
+                        "U",
+                        bspParticipantId,
+                        bspSampleId,
+                        bspMaterialType,
+                        bspReceptacleType);
+            }
+            else {
+                throw new RuntimeException("No participant id for " + kitLabel + " from " + bspKitQueryResult.getInstanceName());
+            }
+        });
+    }
+
+    private static void writeSampleReceivedToES(DDPInstance ddpInstance, BSPKitQueryResult bspKitInfo) {
+        String kitRequestId = new KitRequestDao().getKitRequestIdByBSPParticipantId(bspKitInfo.getBspParticipantId());
+        Map<String, Object> nameValuesMap = new HashMap<>();
+        nameValuesMap.put(ESObjectConstants.RECEIVED, SystemUtil.getISO8601DateString());
+        ElasticSearchUtil.writeSample(ddpInstance, kitRequestId, bspKitInfo.getDdpParticipantId(), ESObjectConstants.SAMPLES,
+                ESObjectConstants.KIT_REQUEST_ID, nameValuesMap);
+    }
+
+    private void triggerDDP(Connection conn, @NonNull BSPKitQueryResult bspKitInfo, boolean firstTimeReceived, String kitLabel) {
+        try {
+            if (bspKitInfo.isHasParticipantNotifications() && firstTimeReceived) {
+                KitDDPNotification kitDDPNotification = KitDDPNotification.getKitDDPNotification(TransactionWrapper.getSqlFromConfig(ApplicationConfigConstants.GET_RECEIVED_KIT_INFORMATION_FOR_NOTIFICATION_EMAIL), kitLabel, 1);
+                if (kitDDPNotification != null) {
+                    EventUtil.triggerDDP(conn, kitDDPNotification);
```

Comment:
cutting connection here would be great as well.

Reply:
I think we would want triggerDDP to happen within the same transaction.

Reply:
what's the point of triggering withing the same transaction?

Reply:
so if anything goes wrong during the whole procedure it is not marked as trigger and that can happen again. at least this is for the UPS test boston kits,  @SimoneMaiwald what do you think?